android-pdf-viewer/src/main/java/com/github/barteksc/pdfviewer/PdfFile.kt

```diff
@@ -0,0 +1,411 @@
+/**
+ * Copyright 2017 Bartosz Schiller
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.barteksc.pdfviewer
+
+import android.graphics.Bitmap
+import android.graphics.Rect
+import android.graphics.RectF
+import android.util.SparseBooleanArray
+import com.github.barteksc.pdfviewer.exception.PageRenderingException
+import com.github.barteksc.pdfviewer.util.FitPolicy
+import com.github.barteksc.pdfviewer.util.PageSizeCalculator
+import org.benjinus.pdfium.Bookmark
+import org.benjinus.pdfium.Link
+import org.benjinus.pdfium.Meta
+import org.benjinus.pdfium.PdfDocument
+import org.benjinus.pdfium.PdfiumSDK
+import org.benjinus.pdfium.util.Size
+import org.benjinus.pdfium.util.SizeF
+
+class PdfFile internal constructor(
```

Comment:
Can you offer some KDoc here since you spent already time with this class regarding its purpose to help first time readers understand what this is about?

Reply:
Done